rfcs/shared/build-system/06-browser-support-for-v9.md

```diff
@@ -0,0 +1,61 @@
+# RFC: Browser Support for Fluent v9
+
+Contributors: Fluent UI React Build team, @TristanWatanabe
+
+## Summary
+
+Fluent v9 is currently lacking a defined browser support matrix so this RFC proposes a set of major browser versions compatible with the library to act as the source of truth. As mentioned below, Fluent developers are expected to conform to these browsers and ensure no incompatible syntax, browser APIs or CSS features are introduced to the library. Consumers who require older browser support that aren't covered in this matrix are responsible for performing downlevel transpilations and/or using polyfills themselves.
```

Comment:
This was intended to say that Fluent wont be **providing** polyfills for features that we expect our browser matrix to already support (i.e. ResizeObserver) - i'll rephrase it

Reply:
I see, what I was trying to point out is that we cannot guarantee this as it cannot be baked by tooling to 100%. So we might end up shipping "some" polyfills.

To give you better context, based on this RFC our build process will look like following:

1. browserlist config with browser matrix
2. babel consuming browserlist and transpiling according to that 

Now one can use ECMA features beyond 2019 and it will just work because babel will transpile that + provide according polyfill from corejs

What about TypeScript? 

well we will still use `target` and `lib` specification - which will be used only for type declarations. Now for production code this might become problematic as target/lib cannot be matched to browser matrix in any way, so we'll end up by "approximate" target boundaries specification on type safety level.

Reply:
Some rephrasing regarding this [ebe76cd](https://github.com/microsoft/fluentui/pull/22889/commits/ebe76cd1b5450a997404a4750fe0147d83035e70)

Reply:
We're seeing some of these same things pop up in MADS (admin-controls) as well in terms of `target` and `lib`. I like the line drawn. We also use nullish coalescing so I think ES2020 is our line. Do you plan to look at that as well? It's very helpful to avoid pesky falsey bugs when looking for values.

Reply:
@brandonthomas We definitely want to support ES2020 features like `nulling coalescing` which is why I ensured this browser matrix supports all ES2020 features so no downlevel transpilation need to occur when we do move our target to ES2020. So in essence, this does draw the line at ES2020 for now